.github/workflows/project_filter.yml

```diff
@@ -0,0 +1,92 @@
+name: CI/CD per project
+on:
+  [ pull_request ]
+
+jobs:
+  run_ci:
+    strategy:
+      matrix:
+        include:
+          - appid: nf01
+            apppath: 01-indoor-floor-level-tracker
+            pio_project: firmware
+            shell_script: ''
+          - appid: nf04
+            apppath: 04-low-power-digital-signage
+            pio_project: firmware
+          - appid: nf08
+            apppath: 08-power-quality-monitor
+            pio_project: firmware
+          - appid: nf09
+            apppath: 09-valve-monitor
+            pio_project: firmware/arduino
+          - appid: nf10
+            apppath: 10-flow-rate-monitor
+            pio_project: firmware
+          - appid: nf11
+            apppath: 11-generator-activity-monitor
+            pio_project: firmware
+          - appid: nf12
+            apppath: 12-remote-power-control
+            pio_project: firmware
+          - appid: nf13
+            apppath: 13-tool-usage-cycle-tracking
+            pio_project: firmware
+          - appid: nf15
+            apppath: 15-greenhouse-monitor
+            pio_project: firmware
+          - appid: nf18
+            apppath: 18-temperature-and-humidity-monitor
+            pio_project: firmware/arduino
+          - appid: nf30
+            apppath: 38-audio-classifier
+            pio_project: classifier
+          - appid: nf30
+            apppath: 38-audio-classifier
+            pio_project: data_acquisition
+          - appid: nf34
+            apppath: 34-cellular-modbus-client
+            pio_project: firmware
+          - appid: nf35
+            apppath: 35-CAN-vehicle-monitor
+            pio_project: firmware
+          - appid: nf36
+            apppath: 36-vending-machine-monitor
+            pio_project: firmware
+          - appid: nf38
+            apppath: 38-audio-classifier
+          - appid: nf41
+            apppath: 41-cold-chain-monitor
+            pio_project: firmware/arduino
+          - appid: nf43
+            apppath: 43-propane-tank-fuel-gauge
+            pio_project: firmware
+          - appid: sense-hat-reference
+            apppath: sense-hat-reference
+            pio_project: .
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: false
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          # uncomment for local testing
```

Comment:
Could we add a short README.md (or similar) explaining how to run this workflow locally?